### PR TITLE
Make operations throw correct SQLException

### LINKED ISSUE #470: Incorrect SQLException handling on Android
There are three classes named `SQLException`:
- `java.sql.SQLException` (checked exception)
- `android.database.SQLException` (unchecked exception)
- `net.sqlcipher.SQLException` (unchecked exception)

The abstract class `SQLDatabase` declares that some methods throw `SQLException` (referring to `java.sql.SQLException`), however, the concrete classes `AndroidSQLite` and `AndroidSQLCipherSQLite` throw the unchecked exceptions `android.database.SQLException` and `net.sqlcipher.SQLException` respectively, which could lead to crashes at runtime for exceptions that on initial inspection appear to be handled.

To fix this, we should catch `android.database.SQLException` and `net.sqlcipher.SQLException` and wrap them in a `java.sql.SQLException` so they behave as expected.

### PR DESCRIPTION
**Pull request by brynh**
Wrap `android.database.SQLException` and `net.sqlcipher.SQLException` in `java.sql.SQLException`.

Fixes #470

### CODE REVIEW

File: cloudant-sync-datastore-android-encryption/src/main/java/com/cloudant/sync/internal/sqlite/android/AndroidSQLCipherSQLite.java [hunk 1]
```diff
@@ -101,19 +102,27 @@ public void setTransactionSuccessful() {
     }
 
     @Override
-    public void execSQL(String sql)  {
+    public void execSQL(String sql) throws java.sql.SQLException {
```
Comment (rhyshort) [inline cloudant-sync-datastore-android-encryption/src/main/java/com/cloudant/sync/internal/sqlite/android/AndroidSQLCipherSQLite.java]: Missed compact on line 68

File: cloudant-sync-datastore-android/src/main/java/com/cloudant/sync/internal/sqlite/android/AndroidSQLite.java [hunk 2]
```diff
@@ -86,15 +86,23 @@ public void setTransactionSuccessful() {
     }
 
     @Override
-    public void execSQL(String sql) throws SQLException {
+    public void execSQL(String sql) throws java.sql.SQLException {
```
Comment (rhyshort) [inline cloudant-sync-datastore-android/src/main/java/com/cloudant/sync/internal/sqlite/android/AndroidSQLite.java]: compact missed here too.

File: cloudant-sync-datastore-android-encryption/src/main/java/com/cloudant/sync/internal/sqlite/android/AndroidSQLCipherSQLite.java (hunk 1 above)
Comment (brynh) [inline cloudant-sync-datastore-android-encryption/src/main/java/com/cloudant/sync/internal/sqlite/android/AndroidSQLCipherSQLite.java]: Good spot. Fixed in 4272ecb.

I've handled it the same way as the java se version in [SQLiteWrapper.java](https://github.com/cloudant/sync-android/blob/master/cloudant-sync-datastore-javase/src/main/java/com/cloudant/sync/sqlite/sqlite4java/SQLiteWrapper.java#L104) so they are consistent.

File: cloudant-sync-datastore-android/src/main/java/com/cloudant/sync/internal/sqlite/android/AndroidSQLite.java (hunk 2 above)
Comment (brynh) [inline cloudant-sync-datastore-android/src/main/java/com/cloudant/sync/internal/sqlite/android/AndroidSQLite.java]: Fixed in 4272ecb.

**Pull request merged**